### ISSUE TITLE
Lack of intellisense when target_arch other than current machine's 

### ISSUE THREAD

**Issue by JakubKoralewski** [intellij-rust/intellij-rust#5183]
## Environment

* **IntelliJ Rust plugin version:** 0.2.118.2157-201
* **IDE name and version:**  CLion 2020.1 EAP
Build #CL-201.6251.14, built on March 11, 2020
Experimental macro resolution is enabled!

## Problem description
![obraz](https://user-images.githubusercontent.com/43069023/78348991-335be180-75a3-11ea-8c23-090e306fce78.png)

I'd like to have code completion for `js_sys::Object`, and other conditionally disabled types somehow. Is there an option to set the `target_arch` somehow to something different than the current machine's?
`js_` ✔️  `js_sys`
`js_sys::Re` ✔️  `js_sys::Reflect`
Even if the above are completed, then on `ctrl+q` they have no documentation, although `ctrl+q` on `js_sys` does show docs

`Object::` :x: empty nothing
`js_sys::Obj` ❌ `js_sys::Object`
`JsValue::fr` ✔️ `JsValue::from_bool` and many more
This is a weird one:
`js_sys::Reflect::se` ✔️ `js_sys::Reflect::set`
`Reflect::se` ❌  `Reflect::set`

## Steps to reproduce

```rust
#[cfg(target_arch = "wasm32")]
use {
	wasm_bindgen::{
		prelude::*,
		JsValue
	},
	js_sys::{
		Object,
		Reflect
	},
}
#[cfg(target_arch = "wasm32")]
#[cfg_attr(target_arch= "wasm32", wasm_bindgen)]
impl SomeStruct {
	#[wasm_bindgen]
	pub fn take(self) -> Object {
		let obj = Object::new();
		
	}
}
```

This has the same effect:
```rust
#[cfg(target_arch = "wasm32")]
#[wasm_bindgen]
impl SomeStruct {
...
}
```

**Comment by ortem** [comment]
@JakubKoralewski You can add these lines to your `.cargo/config` to set `target_arch="wasm32"`: 
```Toml
[build]
target = "wasm32-unknown-unknown"
```

**Comment by Undin** [comment]
@JakubKoralewski 
FYI now the plugin takes Cargo config into account to evaluate `cfg` options. So you need to add the following code into your config
```toml
[build]
target = "wasm32-unknown-unknown"
```

to fix the issue.
Note, the feature is available in [nightly](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-quick-start.html#install-nightly) builds for now and will be landed in the next stable plugin version. Also, it requires at least Rust 1.52 (currently it beta)

**Comment by Miha-Rozina** [comment]
While the `.cargo/config` approach works, it does mean that anytime you want to change the target, you have to rebuild the whole project. It would still be great if we could just tell CLion for which target it should show intellisense without the need to fully rebuild the project.

**Comment by TimonPost** [comment]
Is this still an issue? I do experience the same problem and while in VsCode the rust-analyzer works well in Clion it doesnt test the code within wasm32 targets.

**Comment by TimonPost** [comment]
This issue seems still relevant. It would be nice to just be able to pass a target to rust plugin.

**Comment by cherryleafroad** [comment]
I would like a more normal way of doing this too. Having to switch around the target in cargo configs isn't very ideal and decreases valuable development time

**Comment by nocker01** [comment]
I was about to post a message that even after changing the `.cargo/config` file, the IDE still did not enable the code for the particular target.  However, I found (by chance and frustration) that when I restarted the IDE then the code in the `cfg()` section was enabled as desired.

I am now in the same boat as the last few comments... i.e. it works for me but it would be nice to have a more convenient method of applying this setting.

Thanks.

**Comment by Undin** [comment]
@nocker01 you can reload the project model and apply your changes by manual invocation of `Refresh Cargo Projects` action in Cargo tool window without IDE restart
<img width="310" alt="Screen Shot 2022-07-12 at 08 43 02" src="https://user-images.githubusercontent.com/2539310/178417987-920787d6-b517-4e02-8843-d40845dc3793.png">

Also, after #9012 the plugin detects changes in Cargo config files and can reload the project model itself or notify you about the corresponding changes (depending on your settings)

**Comment by richvdh** [comment]
Something else: in a Cargo workspace, consisting of multiple crates, the `target` has to be set at the workspace level. Setting at the individual crate level is insufficient.

**Comment by siy** [comment]
I have an embedded project, which is built for 3 (soon will be 4) different devices from same set of sources. Devices are differentiated during compilation by setting different "target_os". Obviously, there is no way to derive this "target_os" value from the cargo config target setting. So, it would be extremely convenient to have the ability to set the "target_os" in the IDE.
